service/history/workflow/mutable_state.go

```diff
@@ -95,6 +103,7 @@ type (
 		AddChildWorkflowExecutionTerminatedEvent(int64, *commonpb.WorkflowExecution, *historypb.WorkflowExecutionTerminatedEventAttributes) (*historypb.HistoryEvent, error)
 		AddChildWorkflowExecutionTimedOutEvent(int64, *commonpb.WorkflowExecution, *historypb.WorkflowExecutionTimedOutEventAttributes) (*historypb.HistoryEvent, error)
 		AddCompletedWorkflowEvent(int64, *commandpb.CompleteWorkflowExecutionCommandAttributes) (*historypb.HistoryEvent, error)
+		NewWorkflowForRetryOrCron(int64, *historypb.WorkflowExecutionStartedEventAttributes, *commonpb.Payloads, *failurepb.Failure, time.Duration, NewWorkflowType) (MutableState, error)
```

Comment:
mutable state interface should only contains `actions` not control logic, like `NewWorkflowForRetryOrCron`

Reply:
the name is bad but this is doing basically a subset of `AddContinueAsNewEvent`, which is in MutableState. so maybe that shouldn't be in MutableState either? I can try to move this one out.